### PR TITLE
Refactor Quote Endpoints

### PR DESCRIPTION
**Pull request by Bruception**
### Description

- Added middleware to check configuration for routes.
- Adopted `asyncHandlerWrapper` across the quote endpoints.
- Default to sending a 204 status if headers are not sent and no data is returned by a handler.

### CODE REVIEW

File: backend/api/controllers/quote-ratings.js [hunk 1]
```diff
@@ -3,75 +3,70 @@ const UserDAO = require("../../dao/user");
...
+    let { uid } = req.decodedToken;
+    let { quoteId, rating, language } = req.body;
 
-      rating = Math.round(rating);
+    quoteId = parseInt(quoteId);
+    rating = parseInt(rating);
 
-      //check if user already submitted a rating
-      let user = await UserDAO.getUser(uid);
+    if (isNaN(quoteId) || isNaN(rating)) {
```
Comment (Bruception) [inline backend/api/controllers/quote-ratings.js:19]: Will be moving this validation logic outside.

File: backend/api/controllers/new-quotes.js [hunk 2]
```diff
@@ -5,37 +5,36 @@ const Logger = require("../../handlers/logger.js");
...
-
     return await NewQuotesDAO.get();
   }
 
-  static async addQuote(req, res) {
-    const { uid } = req.decodedToken;
-    const { text, source, language, captcha } = req.body;
-
+  static async addQuote(req, _res) {
+    if (req.context.configuration.quoteSubmit.enabled === false)
```
Comment (Bruception) [inline backend/api/controllers/new-quotes.js:18]: This logic was moved outside of the controller.

File: backend/api/controllers/new-quotes.js [hunk 3]
```diff
@@ -5,37 +5,36 @@ const Logger = require("../../handlers/logger.js");
...
+      throw new MonkeyError(
+        500,
+        "Quote submission is disabled temporarily. The queue is quite long and we need some time to catch up."
+      );
+    let { uid } = req.decodedToken;
```
Comment (Bruception) [inline backend/api/controllers/new-quotes.js:23]: Changed these to `const`.

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**